### PR TITLE
Add citing guide, as per issue #566

### PR DESCRIPTION
**Pull request by jake-87**

### CODE REVIEW

**Comment by sharkdp** [comment]
Thank you!

I just found out that there is a new answer on the stackexchange post that I referenced in the other linked issue: https://academia.stackexchange.com/a/172780

It turns out GitHub actually has builtin support for this. Would it be too much to ask to convert this README section into a `CITATION.cff` file (https://citation-file-format.github.io/)?

File: README.md [hunk 1]
```diff
@@ -283,6 +283,24 @@ The name *hyperfine* was chosen in reference to the hyperfine levels of caesium
...
 
+## Citing hyperfine
+
+sharkdp, hyperfine, (2022), GitHub repository, https://github.com/sharkdp/hyperfine
+
+Or, as a bibtex entry for hyperfine `v1.15.0`:
+
+```
+@misc{sharkdp,
+  author = {sharkdp},
```
Comment (sharkdp) [inline README.md:294]:
```suggestion
  author = {Peter, David},
```

**Comment by sharkdp** [comment]
My orcid is https://orcid.org/0000-0001-7950-9915

**Comment by sharkdp** [comment]
Just found this generator (https://citation-file-format.github.io/cff-initializer-javascript/#/), so here we go:
```cff
cff-version: 1.2.0
title: hyperfine
message: >-
  If you use this software in scientific
  publications, please consider citing it using the
  metadata from this file.
type: software
authors:
  - given-names: David
    family-names: Peter
    email: mail@david-peter.de
    orcid: 'https://orcid.org/0000-0001-7950-9915'
repository-code: 'https://github.com/sharkdp/hyperfine'
abstract: A command-line benchmarking tool.
license: MIT
version: 1.15.0
date-released: '2022-09-07'
```

**Comment by jake-87** [comment]
Thanks! Fixed.

**Comment by jake-87** [comment]
That should provide all information needed to cite hyperfine.

**Review by sharkdp — APPROVED** [review]

**Pull request merged**